simpleclient_spring_boot/src/main/java/io/prometheus/client/spring/EnablePrometheusCollectorRegistration.java

```diff
@@ -0,0 +1,52 @@
+package io.prometheus.client.spring;
+
+import org.springframework.context.annotation.Import;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An {@literal @}Enable style annotation that provides CollectorRegistry related functions.
+ *
+ * <pre><code>
+ *{@literal @}Configuration
+ *{@literal @}EnablePrometheusCollectorRegistration(useDefault = true, register = true)
+ * public class MyConfiguration {
+ *
+ *  {@literal @}Bean
+ *   public Counter counter() {
```

Comment:
Given, a Spring application that processes orders and a Counter that counts them.

If the whole application is started for a test, the first time it uses the static field the Counter is registered in the static CollectorRegistry (using `counter.register()`).

If the JVM is not recycled (it is usually not recycled during builds), the next test that starts the whole application will also try to register the same Counter in the CollectorRegistry - which will fail, because the other Counter is still there - incidentally causing a small memory leak.